lib/src/utils/build_inputs.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:path/path.dart' as p;
+
+/// Builds and returns the list of inputs on which the formatter should be run.
+///
+/// These inputs are determined by expanding the [include] globs and filtering
+/// out any paths that match the expanded [exclude] globs.
+///
+/// Logs may be output in certain scenarios for debugging purposes.
+///
+/// By default these globs are assumed to be relative to the current working
+/// directory, but that can be overridden via [root] for testing purposes.
+Map<String, dynamic> buildInputs(
```

Comment:
Instead of returning a Map, we should create a class to hold the relevant information so that consumers can access it in a type-safe manner rather than relying on particular string keys.

Reply:
We should also rename this util slightly since we're making it public. We want it to be clear that it is intended to be used to gather inputs for a dart formatter tool. Maybe `getFormatterInputs`?